dissect/target/plugins/os/unix/linux/modules.py

```diff
@@ -0,0 +1,71 @@
+from dataclasses import dataclass
+from typing import Iterator
+
+from dissect.target.helpers.record import TargetRecordDescriptor
+from dissect.target.plugin import Plugin, export
+from dissect.target.target import Target
+
+ModuleRecord = TargetRecordDescriptor(
+    "unix/module",
+    [
+        ("string", "name"),
+        ("varint", "size"),
+        ("varint", "refcount"),
+        ("string[]", "used_by"),
+        ("path", "source"),
+    ],
+)
+
+
+@dataclass
+class Module:
+    path: str
+    name: str
+    size: int
+    refcnt: int
+    used_by: list[str]
+
+
+class ModulePlugin(Plugin):
+    def __init__(self, target: Target):
+        super().__init__(target)
+        self._module_paths = list(self.target.fs.path("/sys/module").iterdir())
+
+    def check_compatible(self) -> bool:
+        return len(self._module_paths) > 0
```

Comment:
Check module should raise an `UnsupportedPluginError` instead of returning a `bool`:

```suggestion
    def check_compatible(self) -> None:
        if not len(self._module_paths) > 0:
            raise UnsupportedPluginError("No module paths found.")
```